components/steering_wheel/HW/HW_gpio.c

```diff
@@ -58,3 +58,26 @@ void MX_GPIO_Init(void)
     GPIO_InitStruct.Pull = GPIO_PULLUP;
     HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);
 }
+
+/**
+ * @brief Reads the input pin state
+ * @param[in] input Input pin
+ * @return HW pin state
+ */
+HW_GPIO_PinState_E HW_GPIO_ReadPin(const HW_GPIO_Input_S *input)
+{
+    GPIO_PinState bitstatus;
+
+    /* Check the parameters */
+    assert_param(IS_GPIO_PIN(input->pin));

```

Comment:
we don't need this. This is useful if you're an API maintainer where the API will be used by external folks, but that's not the case here.

Reply:
Should we not consider long term maintainability of these projects internally as well?